File: sdk_python/examples/playback_token.py
"""Sign a viewer playback token and resolve a playback endpoint with it."""

import os

from livepeer_frameworks import FrameWorksClient, sign_playback_token


def main() -> None:
    token = sign_playback_token(
        os.environ["FRAMEWORKS_SIGNING_PRIVATE_KEY"],
        os.environ["FRAMEWORKS_SIGNING_KID"],
        subject="viewer-42",
        audience="viewer",
        expires_in=300,
        claims={"tier": "pro"},
    )
    with FrameWorksClient("https://bridge.frameworks.network/graphql") as fw:
        resolved = fw.resolve_viewer_endpoint(content_id="PLAYBACK_ID", playback_token=token)
        endpoint = resolved.resolve_viewer_endpoint
        if endpoint and endpoint.primary:
            print(f"{endpoint.primary.url}?jwt={token}")


if __name__ == "__main__":
    main()

File: npm_api/examples/playbackToken.ts
// Sign a viewer token for content with a JWT playback policy, and resolve
// the viewer endpoint with it.
import {
  createClient,
  ResolveViewerEndpointDocument,
  signPlaybackToken,
} from "@livepeer-frameworks/api";

const token = await signPlaybackToken({
  privateKeyPem: process.env.FRAMEWORKS_SIGNING_KEY_PEM ?? "",
  kid: process.env.FRAMEWORKS_SIGNING_KID ?? "",
  expiresIn: 300,
  subject: "viewer-42",
  audience: "viewer",
  claims: { tier: "pro" },
});

const client = createClient({ url: "https://bridge.example.com/graphql" });
const data = await client.request(
  ResolveViewerEndpointDocument,
  { contentId: "k3v9x2m7q4w8", protocol: "HLS" },
  { playbackToken: token }
);
// The media URL itself also needs the token: append ?jwt=<token>.
console.log(`${data.resolveViewerEndpoint?.primary?.url}?jwt=${token}`);
